torch/_dynamo/variables/user_defined.py

```diff
@@ -224,6 +226,22 @@ def __str__(self):
     def python_type(self):
         return self.value_type
 
+    def call_torch_function(self, tx, fn, types, args, kwargs):
+        assert has_torch_function(
+            self
+        ), "calling torch function on object without __torch_function__"
+        from .torch_function import _get_subclass_type_var, call_torch_function
+
+        return call_torch_function(
+            tx,
+            _get_subclass_type_var(tx, self),
+            self.torch_function_fn,
+            fn,
+            types,
+            args,
+            kwargs,
+        )
```

Comment:
This is rather hard to follow, can you explain what this is doing?

Reply:
`call_torch_function` is the main entry point to inlining a `__torch_function__` impl, each class that supports `__torch_function__` arranges the arguments according to the `__torch_function__` signature and calls that to inline through the torch function var.